course02/homework01/gavin/list.cpp

```diff
@@ -0,0 +1,143 @@
+#include <memory>
+
+#include <memory>
+
+#include "list.h"
+
+void List::clear() {
+    std::unique_ptr<Element> newList;
+    mFirst = std::move(newList);
```

Comment:
Wow, that's an interesting way to reset your shared pointer! Why constructing a unique_pointer... and moving it? Why not just resetting your shared pointer?